src/lib/twizzler-driver/src/dma/pool.rs

```diff
@@ -133,6 +191,37 @@ impl BaseType for EmptyBase {
     }
 }
 
+// Merge adjacent regions by sorting, comparing pairs, and merging if they are adjacent.
```

Comment:
Not in this commit, but...is there a good way to write a perf test for this?  It's a key bit of code and I'm not sure I trust the default sorting stuff to be fast.

Reply:
Hmm, yeah, this is probably a good idea. We can open an issue for it and tie it to the RFC or something.